graph-manager-docs/source/schema-validation.mdx

```diff
@@ -1,36 +1,200 @@
+      # variable. Configure your endpoint's location in your Apollo config.
+      - run: npx apollo service:check
+```
+
+### Integrating with GitHub
+
+If you're using GitHub, you can install the [Apollo Engine GitHub app](https://github.com/apps/apollo-engine). This app enables Graph Manager to send a webhook back to your project on each call to `apollo service:check`, providing built-in pass/fail status checks on your pull requests:
+
+![GitHub Status View](./img/schema-validation/github-check.png)
+
+> **Note:** If you're using GitHub status checks, we recommend _ignoring_ the exit code of the `apollo service:check` command so your continuous integration can complete without failing early. You can do this by appending `|| echo 'validation failed'` to the command call.
+
+### Integrating with other version control services
+
+If you're using GitHub Enterprise, Bitbucket, or another version control service, we recommend setting up your CI tool to post a comment on each pull request with the results of schema validation. By surfacing schema diffs and breaking changes directly in your PR, you can avoid having to search your CI logs to check why validation failed.
+
+The `apollo service:check` supports a `--markdown` flag, which outputs the results of schema validation in markdown format. You can pipe this output directly into a comment to your source control tool, like in [this example of posting a comment with the results of schema validation to GitHub](https://gist.github.com/daniman/e53d0589d18b778878bd8ef32d2e793c).
+
+The output of `apollo service:check --markdown` looks like this:
+
+```md
+### Apollo Service Check
+
+🔄 Validated your local schema against schema tag `staging` on service `engine`.
+🔢 Compared **18 schema changes** against **100 operations** seen over the **last 24 hours**.
+❌ Found **7 breaking changes** that would affect **3 operations** across **2 clients**
+
+🔗 [View your service check details](https://engine.apollographql.com/service/engine/checks?...).
+```
+
+## Validating against multiple environments
+
+You might want to validate your schema changes against multiple environments, such as production, staging, and beta. Each of these environments might have a slightly different schema to support features that are experimental or in active development. In Apollo Graph Manager, these schemas typically belong to [variants](./schema-registry/#managing-environments-with-variants) of a single data graph.
+
+You can validate your schema against a particular variant by including the `--tag` flag in your call to `apollo service:check`, like so:
+
+```
+$ apollo service:check --tag=staging
+```
+
+If you want to validate your schema against multiple variants, call `apollo service:check` once for each variant. Running `service:check` against multiple variants results in status checks similar to the following:
+
+![multiple service checks](./img/schema-validation/multi-github-check.png)
+
+## Customizing validation
```

Comment:
We should probably note that if you pass in a validationPeriod outside of your account's retention, then it will error.